src/vos/vos_container.c

```diff
@@ -532,31 +528,30 @@ vos_cont_destroy(daos_handle_t poh, uuid_t co_uuid)
 		}
 	}
 
-	rc = cont_df_lookup(vpool, &key, &args);
+
+	rc = cont_df_lookup(pool, &key, &args);
 	if (rc) {
 		D_DEBUG(DB_TRACE, DF_UUID" container does not exist\n",
 			DP_UUID(co_uuid));
 		D_GOTO(exit, rc);
 	}
 
-	rc = vos_tx_begin(vpool);
-	if (rc != 0)
-		goto failed;
-
-	rc = vos_obj_tab_destroy(vpool, &args.ca_cont_df->cd_otab_df);
+	rc = vos_tx_begin(pool);
 	if (rc) {
-		D_ERROR("OI destroy failed with error : %d\n", rc);
-		goto end;
+		D_ERROR("Failed to start pmdk transaction: %s\n", d_errstr(rc));
+		D_GOTO(exit, rc);
 	}
 
 	d_iov_set(&iov, &key, sizeof(struct d_uuid));
-	rc = dbtree_delete(vpool->vp_cont_th, BTR_PROBE_EQ, &iov, NULL);
+	rc = dbtree_delete(pool->vp_cont_th, BTR_PROBE_EQ, &iov, NULL);
 
-end:
-	rc = vos_tx_end(vpool, rc);
-failed:
-	if (rc != 0)
-		D_ERROR("Destroying container transaction failed %d\n", rc);
+	rc = vos_tx_end(pool, rc);
+	if (rc) {
+		D_ERROR("Failed to end pmdk transaction: %s\n", d_errstr(rc));
+		D_GOTO(exit, rc);
+	}
+	gc_wait_pool(pool);
```

Comment:
After container destroy, we need to force VEA expire the just freed extents and call unmap against them and make them available for allocation immediately. I'll cook separate patch for this soon.